tests/helix/send-to-helix-ci.proj

```diff
@@ -4,11 +4,22 @@
       <TestCategory Include="basictests" />
       <TestCategory Condition="'$(OS)' != 'Windows_NT'" Include="endtoendtests" />
 
+      <AdditionalProperties Condition="'$(Configuration)' != ''" Include="Configuration=$(Configuration)" />
+
       <_ProjectsToBuild Include="send-to-helix-inner.proj"
                         Condition="'%(TestCategory.Identity)' != ''"
-                        AdditionalProperties="TestCategory=%(TestCategory.Identity);Configuration=$(Configuration)" />
+                        AdditionalProperties="TestCategory=%(TestCategory.Identity);@(AdditionalProperties, ';')" />
     </ItemGroup>
 
+    <!-- Invoke 'PrepareDependencies' targets with BuildInParallel=false -->
+    <MSBuild Projects="@(_ProjectsToBuild)" Targets="PrepareDependencies" BuildInParallel="false" StopOnFirstFailure="true" />
+
+    <ItemGroup>
+      <!-- Change the set of properties in the evaluated project, so msbuild will build the
+           following projects in parallel, instead of combining with the previous one to build
+           sequentially. -->
+      <_ProjectsToBuild AdditionalProperties="%(AdditionalProperties);__RandomPropertyForDifferentiatingBuilds=$([System.Guid]::NewGuid())" />
```

Comment:
This is weird. Why exactly is this needed?

Reply:
If I have:
```xml
      <_ProjectsToBuild Include="send-to-helix-inner.proj"
                        Condition="'%(TestCategory.Identity)' != ''"
                        AdditionalProperties="TestCategory=%(TestCategory.Identity);@(AdditionalProperties, ';')" />
 
<!-- meant to not run in parallel -->
 <MSBuild Projects="@(_ProjectsToBuild)" Targets="PrepareDependencies" BuildInParallel="false" StopOnFirstFailure="true" />

<!-- meant to run in parallel -->
<MSBuild Projects="@(_ProjectsToBuild)" Targets="Test" BuildInParallel="true" StopOnFirstFailure="false" />
```

With two test categories `basictests;endtoendtests` then msbuild runs:

1. build `PrepareDependencies` for the first category; then build `Test` for the same, and wait till it is done
2. Then repeat (1) but for the second category

It's possibly some optimization in msbuild that combines these two `msbuild` task invocation, and does not run them in parallel. But adding a new property to `AdditionalProperties` means that the two runs cannot be combined. The value of the property can be anything.